src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java

```diff
@@ -57,12 +59,18 @@ private void initializeDrivingFeatures() {
         final GATKPath drivingPath = getDrivingFeaturePath();
         final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(drivingPath.toPath());
         if (isAcceptableFeatureType(codec.getFeatureType())) {
-            drivingFeatures = new FeatureDataSource<>(new FeatureInput<>(drivingPath), FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES, null, cloudPrefetchBuffer, cloudIndexPrefetchBuffer, referenceArguments.getReferencePath());
-
-            final FeatureInput<F> drivingFeaturesInput = new FeatureInput<>(drivingPath, "drivingFeatureFile");
-            features.addToFeatureSources(0, drivingFeaturesInput, codec.getFeatureType(), cloudPrefetchBuffer, cloudIndexPrefetchBuffer,
-                                         referenceArguments.getReferencePath());
-            header = getHeaderForFeatures(drivingFeaturesInput);
+            final GenomicsDBOptions options = new GenomicsDBOptions(referenceArguments.getReferencePath());
+            final FeatureInput<F> drivingFeatureInput = new FeatureInput<>(drivingPath);
+            drivingFeatureInput.setFeatureCodecClass((Class<FeatureCodec<F, ?>>)codec.getClass());
+            drivingFeatures = new FeatureDataSource<>(drivingFeatureInput, FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES, null,
+                    cloudPrefetchBuffer, cloudIndexPrefetchBuffer, options, false);
+            header = drivingFeatures.getHeader();
+
+            final FeatureInput<F> featureInput = new FeatureInput<>(drivingPath, "drivingFeatureFile");
+            featureInput.setFeatureCodecClass((Class<FeatureCodec<F, ?>>)codec.getClass());
+            features.addToFeatureSources(featureInput,
+                    new FeatureDataSource<>(featureInput, FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES, null,
```

Comment:
The previous implementation passed in 0 here for `queryLookaheadBases` rather than `FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES`, and used `FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES` only for `drivingFeatures` itself (above). This was because we know that `drivingFeatures` is going to be used for a full linear single-pass traversal, so the caching makes sense, whereas the secondary "drivingFeatureFile" `FeatureInput` might be used for things like windowed queries around individual loci that would interact poorly with the very simplistic forward-looking caching scheme.

Now, I don't object to turning the caching on in both places if you've done some benchmarking and concluded that doing so will result in better performance for typical `FeatureWalkers` -- is that in fact the case?

Reply:
Nah.  Good catch.  The change was inadvertent.